### PR TITLE
ui: refactor SecretFormShow and SecretCreateOrUpdate

### PR DESCRIPTION
**Pull request by hellobontempo**

### CODE REVIEW

File: ui/app/components/secret-create-or-update.js [hunk 1]
```diff
@@ -138,28 +104,7 @@ export default class SecretCreateOrUpdate extends Component {
           delete secret.selectedVersion.secretData;
         }
         if (!secretData.isError) {
-          if (isV2) {
```
Comment (kiannaquach) [inline ui/app/components/secret-create-or-update.js:141]: so glad we can get to remove this V2 logic!!

File: ui/app/templates/components/secret-edit.hbs [hunk 2]
```diff
@@ -69,16 +69,11 @@
         @showAdvancedMode={{this.showAdvancedMode}}
         @modelForData={{this.modelForData}}
         @error={{this.error}}
-        @isV2={{this.isV2}}
-        @secretData={{this.secretData}}
         @buttonDisabled={{this.buttonDisabled}}
-        @canUpdateSecretMetadata={{this.canUpdateSecretMetadata}}
         @canReadSecretData={{this.canReadSecretData}}
-        @canReadSecretMetadata={{this.canReadSecretMetadata}}
```
Comment (kiannaquach) [inline ui/app/templates/components/secret-edit.hbs:77]: Should we remove `isV2`, `secretData`, `canUpdateSecretMetadata`, `canReadSecretMetadata` in the `secret-edit` component file too?

Comment (hellobontempo) [reply ui/app/templates/components/secret-edit.hbs:77]: I was just trying to break some of these tasks up into smaller ones by starting with the child components of this file - but yes! That will come!

**Comment by hellobontempo** [comment]
Closing this attempt as it broke this test and I can't figure out why!

**Pull request closed without merging**